### ISSUE TITLE
Issue with creating QApplication in QT backend

### ISSUE THREAD

**Issue by macdems** [matplotlib/matplotlib#15100]
In the file [backend_qt5.py in line 126](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/backends/backend_qt5.py#L126) a new `QApplication` object is created. However, the passed argument is a list with a binary string. This causes PySide2 to print ugly

```
QSettings::value: Empty key passed
QSettings::value: Empty key passed
```
Correctly, the string should be passed, which will make the warning go away.

To resolve this issue, please replace

```python
qApp = QtWidgets.QApplication([b"matplotlib"])
```

to this

```python
qApp = QtWidgets.QApplication(["matplotlib"])
```

**Comment by ImportanceOfBeingErnest** [comment]
The confusing bit here is that the `b` was introduced in https://github.com/matplotlib/matplotlib/pull/9169 to actually fix pyside.

**Comment by tacaswell** [comment]
I guess we need some conditional logic on what to pass as the name based on which Qt wrapper we are using.  Want to take a crack at that @macdems ?

**Comment by macdems** [comment]
I have investigated the issue and it seems that for #9162, the critical commit is fb6fa1d07f3448ed32e3690237665b206948ba03. You can revert 52eb8b8397d26827305fd291d8914ad1b29e8936 and the bug #9162 is still not present.

**Comment by macdems** [comment]
But I have checked the code for reproduction in the original issue and it works with PySide and Qt4.

**Comment by macdems** [comment]
Ok, but the problem only happens in Python2 (see the `u"Hello"` in https://bugreports.qt.io/browse/PYSIDE-75). It is not present in Python3. And in Python2 , the string withou the `b` in the beginning is not a unicode, so you can safely remove it. Check that the following code works with both Python versions:

```python
from PySide.QtGui import QApplication
QApplication(["Hello"])
```

Passing `b"matplotlib"` to QApplication in Python3, passes bytes, not string, which is incorrect.

**Comment by anntzer** [comment]
The last commit to pyside dates back to October 2015 (https://github.com/pyside/PySide) which is more than a year earlier than the release of the oldest supported Python (Py 3.6, December 2016 https://www.python.org/downloads/release/python-360/).  Given that we can't even test it properly, I think it's time to call it quits on pyside(1)...

**Comment by macdems** [comment]
I suggest that you simply change `b"matplotlib"` to `"matplotlib" and check that it works in any case.

However, if you insist on some verification, you can make the line look like this:

```python
app = QtWidgets.QApplication(["matplotlib" if isinstance("mpl", str) else b"matplotlib"])
```

This will guard you against the unicode literals (```from __future__ import unicode_literals```), although I do not see them used in ` backend_qt5.py`.

**Comment by tacaswell** [comment]
We do not support python2 so that conditional will always be True.

I think the question we need answered is: Is there a version of `pyside` that works with py36 and requires a bytes object as input?  

Trading suppressing a warning for a segfault seems like a bad deal ;)

**Comment by anntzer** [comment]
Both conda-forge and gohlke have pyside for Py3.6 (https://anaconda.org/conda-forge/pyside/files, https://www.lfd.uci.edu/~gohlke/pythonlibs/#pyside), but not Py3.7.  So I guess it may be reasonable to drop support for it at the same time as we drop support for Py3.6, which according to my calculations should be around mpl3.3.

**Comment by macdems** [comment]
> I think the question we need answered is: Is there a version of `pyside` that works with py36 and requires a bytes object as input?

There is no such version to my knowledge. According to my tests [PYSIDE-75](https://bugreports.qt.io/browse/PYSIDE-75) is only Python2 issue. Actually you must always pass an `str` object and in Python2 `b"matplotlib"` is such an object. In Python3 it is not.

Actually, in Python3, passing `b"matplotlib"` is a bug. In both PySide and PySide2 result in the `app.arguments()` returning `['']`, which is not correct (it should return `['matplotlib']` and this is what both PyQt4 and PyQt5 do). I don't think this really matters in this particular use case, but it is not correct anyway.

If you insist on some workaround (which I believe should not be done), you can always do:

```python
app = QtWidgets.QApplication([b"matplotlib" if 'PySide' in sys.modules else "matplotlib"])
```

**Comment by tacaswell** [comment]
Great, if these is a python2 only work around, then :+1: on that change.  Could you open a PR?

**Comment by tacaswell** [comment]
closed by #15122
